src/lib.rs

```diff
@@ -215,14 +220,44 @@ fn respond_to_request(stream: &mut TcpStream, build_dir_path: impl AsRef<Path>)
 }
 
 #[derive(Debug, StructOpt)]
-pub struct Watch {}
+pub struct Watch {
+    #[structopt(long = "watch", short = "w")]
+    watch_paths: Vec<PathBuf>,
+    #[structopt(long = "ignore", short = "i")]
+    exclude_paths: Vec<PathBuf>,
+}
 
 impl Watch {
-    pub fn execute(
-        &self,
-        watch_path: impl AsRef<Path>,
-        mut command: process::Command,
-    ) -> Result<()> {
+    pub fn new() -> Self {
+        Self {
+            exclude_paths: Vec::new(),
+            watch_paths: Vec::new(),
+        }
+    }
+
+    pub fn add_exclude_path(&mut self, path: impl AsRef<Path>) {
+        self.exclude_paths.push(path.as_ref().to_path_buf())
+    }
+
+    pub fn add_watch_path(&mut self, path: impl AsRef<Path>) {
+        self.watch_paths.push(path.as_ref().to_path_buf())
+    }
+
+    pub fn is_excluded_path(&mut self, path: &Path) -> bool {
+        self.exclude_paths
+            .iter()
+            .map(|x| path.starts_with(x))
+            .any(|x| x)
+    }
+
+    pub fn is_hidden_path(&mut self, path: &Path) -> bool {
+        path.file_name()
+            .and_then(|x| x.to_str())
+            .map(|x| x.starts_with('.'))
+            .unwrap_or(false)
+    }
```

Comment:
The user don't need this. Make it private by removing `pub`.
```suggestion
    fn is_excluded_path(&mut self, path: &Path) -> bool {
        self.exclude_paths
            .iter()
            .map(|x| path.starts_with(x))
            .any(|x| x)
    }

    fn is_hidden_path(&mut self, path: &Path) -> bool {
        path.file_name()
            .and_then(|x| x.to_str())
            .map(|x| x.starts_with('.'))
            .unwrap_or(false)
    }
```